src/app/components/main-list/main-list.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, OnInit } from '@angular/core';
+import { AdsService } from '../../services/ads.service';
+
+@Component({
+  selector: 'app-main-list',
+  templateUrl: './main-list.component.html',
+  styleUrls: ['./main-list.component.scss']
+})
+export class MainListComponent implements OnInit {
+
+  timer: any;
+
+  songList=[];
+  adList = [];
+  index = 1;
+  songs = [{
+    name: 'Varona',
+    artist: 'Geographer'
+  },
+  {
+    name: 'Kite',
+    artist: 'Geographer'
+  }]
+
+
+
+  constructor(private adsService: AdsService) { }
+
+  ngOnInit() {
+    this.adsService.getAds().subscribe((ads: any[]) => {
+      console.log('ADS',ads);
+      this.adList = ads;
+    })
+    this.timer = setInterval(() => {
+      //this is just a filler action.  We would actually so something like
+      // this.adUrl = this.ads[this.index].url
+        console.log(this.songs[this.index].name)
+        if (this.index === this.songs.length -1) {
+        // if we are at the end of the list start again at the beginning;
+          this.index = 0;
+        } else {
+        // else go to next item
+          this.index ++;
+        }
+    }, 4000)
```

Comment:
Now that we have some ad data we should utilize it.  To do that we need to bring the timer assignment into the subscribe and tweak the strucure of our ad data.  Eventually we will have the ad configuration with an id, frequency and duration that will be related to an ad with id, config_id, and image_path (or image_id that gets us a path).  For now we should just have all the urls be an array on the ad data structure
```suggestion
    this.adsService.getAds().subscribe((ads: any[]) => {
      //ads is a list 500 long so maybe only grab the first one for development
      this.adList = ads[0].images; //images is a placeholder name, it may end up as something else
      this.timer = setInterval(() => {
      this.adUrl = this.adList[this.index]
        if (this.index === this.songs.length -1) {
          this.index = 0;
        } else {
          this.index ++;
        }
    }, ads[0].duration* 1000)
    })
    
```